setup.py

```diff
@@ -1,100 +1,176 @@
 
+PACKAGE = 'gourmet'
+PACKAGEDIR = Path('src') / PACKAGE
+DATADIR = Path('data')
+PODIR = Path('po')
+LANGS = sorted(f.stem for f in PODIR.glob('*.po'))
+LOCALEDIR = PACKAGEDIR / 'data' / 'locale'
 
-def modir(lang):
-    mobase = Path("build")
-    return mobase / "mo" / lang
 
+def get_info(prop: str) -> str:
+    with open(PACKAGEDIR / 'version.py') as versfile:
+        content = versfile.read()
+    match = re.search(r'^{} = "(.+)"'.format(prop), content, re.M)
+    if match is not None:
+        return match.group(1)
+    raise RuntimeError(f"Unable to find {prop} string")
 
-def mkmo(lang):
-    outpath = modir(lang)
-    os.makedirs(outpath, exist_ok=True)
 
-    inpath = podir / (lang + ".po")
+def refresh_metadata(cmd: Command) -> None:
+    # distutils remembers what commands are run during a build so that
+    # subsequent calls to the command are skipped. This means that metadata is
+    # normally only created once even if multiple builds are being done (e.g.,
+    # 'python setup.py sdist bdist_wheel'). Since package data is dynamically
+    # built depending on the type of build, we need to ensure that any stale
+    # metadata is refreshed.
+    if cmd.distribution.have_run.get('egg_info', 0):
+        cmd.distribution.reinitialize_command('egg_info')
+        cmd.run_command('egg_info')
 
-    cmd = f"msgfmt {inpath} -o {outpath}/{package}.mo"
-    os.system(cmd)
 
+def rmfile(filepath: Union[Path, str]) -> None:
+    try:
+        os.remove(filepath)
+    except FileNotFoundError:
+        pass
 
-def merge_i18n():
-    cmd = "LC_ALL=C intltool-merge -u -c ./po/.intltool-merge-cache ./po"
 
-    for infile in Path('.').rglob('*.in'):
-        outfile = Path(str(infile)[:-3])
-        extension = outfile.suffix
+class BuildI18n(Command):
+    description = "Build localized message catalogs"
+    user_options = []
+
+    def initialize_options(self):
+        # Command subclasses must implement this "abstract" method
+        pass
+
+    def finalize_options(self):
+        # Command subclasses must implement this "abstract" method
+        pass
+
+    def run(self):
+        # compile message catalogs to binary format
+        for lang in LANGS:
+            pofile = PODIR / f'{lang}.po'
+
+            mofile = LOCALEDIR / lang / 'LC_MESSAGES'
+            mofile.mkdir(parents=True, exist_ok=True)
+            mofile /= f'{PACKAGE}.mo'
+
+            cmd = f'msgfmt {pofile} -o {mofile}'
+            os.system(cmd)
 
-        if 'desktop' in extension:
-            flag = '-d'
-        elif 'schema' in extension:
-            flag = '-s'
-        elif 'xml' in extension:
-            flag = '-x'
-        elif 'gourmet-plugin' in extension:
-            flag = '-k'
-        else:
-            flag = ''
+        # merge translated strings into various file types
+        cachefile = PODIR / '.intltool-merge-cache'
+        cmd = f"LC_ALL=C intltool-merge -u -c {cachefile} {PODIR}"
```

Comment:
Do you know why `LC_ALL=C` environment variable is set here? Is this needed?

Reply:
This is used to force English as the source locale/encoding, to be translated to other locales.

gettext does not necessarily translate from English to to other languages, and application may have different "base" languages or encodings.  

There's more details here: https://www.gnu.org/software/gettext/FAQ.html#nonascii_strings

I think that it's overkill for us, since we have a pretty basic usage of translation.

It's not strictly needed, and I'm neutral on keeping or removing the variable.

Reply:
I'm not sure I understand the implications here, even after reading the link and doing some other googling around. My experience in i18n/l10n is limited to what I've done on this project. LOL. I guess I'm fine leaving it.